### DISCUSSION TITLE
Querying the API wiht both search query and date range

### DISCUSSION THREAD

**Discussion by yopletweets** [O365/python-o365#933] [Q&A]
Tried to query mailboxes with both search query and date range and the API ignores the date range.

Here's the code I used;

```
query:Query = mailbox.new_query()
query.search("('trumo' or 'biden')")
query.chain('and').on_attribute('created_date_time').greater(datetime.datetime(2020, 3, 21))
query.chain('and').on_attribute('created_date_time').less(datetime.datetime(2021, 3, 21))
        
```

**Comment by alejcas** [comment]
MS Graph is very picky and obscure (both) with what search can be done.

Try same search in MS Graph explorer. If it works there it should work here.
